src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -25,6 +27,7 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 
+@Singleton
```

Comment:
Question: Why make this a Singleton? And if it is now a singleton, do you still need the member variables to be 'static'?